ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSwindow.cpp

```diff
@@ -274,227 +279,151 @@ void window_default(bool center_size)
+  // The window should be a maximized popup window.
+  if ((style & WS_POPUP) != WS_POPUP) return false;
+  if ((style & WS_MAXIMIZE) != WS_MAXIMIZE) return false;
+  // The window should have no caption, resize box, or (probably?) border
+  if ((style & WS_CAPTION) == WS_CAPTION) return false;
+  if ((style & WS_THICKFRAME) == WS_THICKFRAME) return false;
+  if ((style & WS_BORDER) == WS_BORDER) return false;
+  return true;
 }
 
-void window_set_minimized(bool minimized)
-{
-    ShowWindow(enigma::hWnd,(minimized?SW_MINIMIZE:SW_RESTORE));
+void window_set_sizeable(bool sizeable) {
+  SetWindowLongPtr(enigma::hWnd, GWL_EXSTYLE,
+      GetWindowLongPtr(enigma::hWnd, GWL_EXSTYLE) | WS_SIZEBOX);
 }
 
-void window_set_maximized(bool maximized)
-{
-    ShowWindow(enigma::hWnd,(maximized?SW_MAXIMIZE:SW_RESTORE));
+bool window_get_sizeable() {
+  return (GetWindowLongPtr(enigma::hWnd, GWL_EXSTYLE) & WS_THICKFRAME) == WS_THICKFRAME;
 }
 
-bool window_get_minimized()
-{
-    return IsIconic(enigma::hWnd);
+void window_set_showborder(bool show) {
+  SetWindowLongPtr(enigma::hWnd, GWL_EXSTYLE,
+      GetWindowLongPtr(enigma::hWnd, GWL_EXSTYLE) | WS_EX_LAYERED);
 }
 
-bool window_get_maximized()
-{
-    return IsZoomed (enigma::hWnd);
+bool window_get_showborder() {
+  return (GetWindowLongPtr(enigma::hWnd, GWL_EXSTYLE) & WS_CAPTION) == WS_CAPTION;
 }
 
-void window_set_stayontop(bool stay)
-{
-    if (enigma::stayOnTop == stay)
-        return;
-
-    if ((enigma::stayOnTop = stay))
-    {
-        SetWindowPos(enigma::hWnd,HWND_TOPMOST,0,0,0,0,SWP_NOSIZE|SWP_NOMOVE);
-    }
-    else
-    {
-        SetWindowPos(enigma::hWnd,HWND_BOTTOM,0,0,0,0,SWP_NOSIZE|SWP_NOMOVE|SWP_NOACTIVATE);
-        SetWindowPos(enigma::hWnd,HWND_TOP,0,0,0,0,SWP_NOSIZE|SWP_NOMOVE);
-    }
+void window_set_showicons(bool show) {
+  SetWindowLongPtr(enigma::hWnd, GWL_EXSTYLE,
+      GetWindowLongPtr(enigma::hWnd, GWL_EXSTYLE) | WS_EX_LAYERED);
 }
 
-void window_set_freezeonlosefocus(bool freeze)
-{
-    enigma::freezeOnLoseFocus = freeze;
+bool window_get_showicons() {
+  return (GetWindowLongPtr(enigma::hWnd, GWL_EXSTYLE) & WS_THICKFRAME) == WS_THICKFRAME;
 }
 
-bool window_get_freezeonlosefocus()
-{
-    return enigma::freezeOnLoseFocus;
+void window_set_visible(bool visible) {
+  ShowWindow(enigma::hWnd, visible? SW_SHOW : SW_HIDE);
 }
 
-bool window_get_stayontop()
-{
-    return enigma::stayOnTop;
+int window_get_visible() {
+  return IsWindowVisible(enigma::hWnd);
 }
 
-void window_set_region_scale(double scale, bool adaptwindow)
-{
-    enigma::viewScale = int(scale*100);
-    enigma::windowAdapt = adaptwindow;
-    enigma::setwindowsize();
+void window_set_minimized(bool minimized) {
+  ShowWindow(enigma::hWnd, minimized? SW_MINIMIZE : SW_RESTORE);
 }
 
-double window_get_region_scale()
-{
-    return enigma::viewScale/100.0;
+void window_set_maximized(bool maximized) {
+  ShowWindow(enigma::hWnd, maximized? SW_MAXIMIZE : SW_RESTORE);
```

Comment:
No, this will override `window_set_visible`, do not call ShowWindow anywhere except `window_set_visible` itself. You should call `SetWindowLongPtr(enigma::hWnd, GWL_STYLE, WS_MAXIMIZE);` instead without the WS_POPUP style. WS_MAXIMIZE only makes the window fullscreen when WS_POPUP style is present also. Do the same for `window_set_minimized` directly above this.